packages/teraslice/test/lib/cluster/services/cluster/backends/kubernetes/k8sResource-spec.js

```diff
@@ -0,0 +1,401 @@
+'use strict';
+
+const _ = require('lodash');
+const yaml = require('js-yaml');
+
+const K8sResource = require('../../../../../../../lib/cluster/services/cluster/backends/kubernetes/k8sResource');
+
+describe('k8sResource', () => {
+    let execution;
+    let terasliceConfig;
+
+    beforeEach(() => {
+        terasliceConfig = {
+            shutdown_timeout: 60000,
+            assets_directory: '',
+            assets_volume: '',
+            name: 'ts-dev1',
+            kubernetes_image: 'teraslice:k8sdev',
+            kubernetes_namespace: 'ts-dev1',
+            kubernetes_image_pull_secret: ''
+        };
+        execution = {
+            name: 'example-data-generator-job',
+            workers: 2,
+            job_id: '7ba9afb0-417a-4936-adc5-b15e31d1edd1',
+            ex_id: 'e76a0278-d9bc-4d78-bf14-431bcd97528c',
+            // slicer_port: 45680,
+            // slicer_hostname: 'teraslice-execution-controller-e76a0278-d9bc-4d78-bf14-431bcd97'
+        };
+    });
+
+    describe('worker deployment', () => {
+        it('has valid resource object.', () => {
+            const kr = new K8sResource(
+                'deployments', 'worker', terasliceConfig, execution
+            );
+
+            expect(kr.resource.kind).toBe('Deployment');
+            expect(kr.resource.spec.replicas).toBe(2);
+            expect(kr.resource.metadata.name).toBe('ts-wkr-example-data-generator-job-7ba9afb0-417a');
+
+            // The following properties should be absent in the default case
+            expect(kr.resource.spec.template.spec).not.toHaveProperty('affinity');
```

Comment:
`expect().toHaveProperty()` works with dot notation, so this can be simplified to 
```suggestion
            expect(kr).not.toHaveProperty('resource.spec.template.spec.affinity');
```